cmds/fmap/lib/fmap.go

```diff
@@ -22,15 +22,19 @@ const (
 	FmapAreaReadOnly
 )
 
-type FMap struct {
-	Start     uint64 // not part of the written struct, but useful nevertheless
+type fMapInternal struct {
 	Signature [8]uint8
 	VerMajor  uint8
 	VerMinor  uint8
 	Base      uint64
 	Size      uint32
 	Name      [32]uint8
```

Comment:
My line of thinking is that `nAreas` is already encoded in `len(Areas)` since Go stores the slice length with the slice. With an `nAreas` field, manipulating the `Areas` slice always requires two steps:

    fmap.Areas = append(fmap.Areas, newArea)
    fmap.nAreas++

Instead, the struct is normalized and only the first line is need. The to-be-implemented `WriteFmap` function will have `binary.Write(writer, uint16(len(fmap.Areas)))` in place of writing `nAreas`.

Either way could work, but I prefer the current implementation. It makes it impossible for `fmap.nAreas` and `len(fmap.Areas)` to become out-of-sync.